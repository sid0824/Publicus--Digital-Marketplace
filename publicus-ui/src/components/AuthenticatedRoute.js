import React from "react";
import { Route, Redirect } from "react-router-dom";
import {withAuthContext} from "../AuthContext";
import { message } from 'antd';

// Renders the redirect components that redirects to the supplied path
const redirect = (path) => {
  message.warning('Login With an Account To Access this Page');
  return <Redirect
    to={path} // TODO - make dynamic based on url parameters
  />
};

// Used to redirect the user away from pages that require an account. Redirects to the login page with a warning message
export const AuthenticatedRoute = ({ component: Component, authenticated, setUserHasAuthenticated, ...rest }) =>
  <Route
    {...rest}
    render={props =>
      authenticated
        ? <Component {...props} />
        : redirect(`/login`)
    }
  />;

export default withAuthContext(AuthenticatedRoute)
