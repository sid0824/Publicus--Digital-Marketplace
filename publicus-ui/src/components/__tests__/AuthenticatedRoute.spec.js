import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import { message } from 'antd';
import Adapter from 'enzyme-adapter-react-16';
import { AuthenticatedRoute } from "../AuthenticatedRoute";

configure({ adapter: new Adapter() });

jest.mock('antd');

describe('Authenticated Route', () => {

  beforeEach(() => {

  });

  afterAll(() => {

  });

  test('should render authorised authenticated route', () => {
    const wrapper = shallow(
      <AuthenticatedRoute
        component={<div>Component</div>}
        authenticated
        other={'props'}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should redirect on un-authorised authenticated route', () => {
    const wrapper = shallow(
      <AuthenticatedRoute
        component={<div>Component</div>}
        authenticated={false}
        other={'props'}
      />
    );
    const redirect = wrapper.props().render({ other: 'value '});
    expect(message.warning).toHaveBeenCalledWith('Login With an Account To Access this Page');
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(shallow(redirect))).toMatchSnapshot();
  });


});