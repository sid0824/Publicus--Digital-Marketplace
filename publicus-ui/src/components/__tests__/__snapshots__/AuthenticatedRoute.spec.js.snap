// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticated Route should redirect on un-authorised authenticated route 1`] = `
<Route
  other="props"
  render={[Function]}
/>
`;

exports[`Authenticated Route should redirect on un-authorised authenticated route 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Authenticated Route should render authorised authenticated route 1`] = `
<Route
  other="props"
  render={[Function]}
/>
`;
